projects/step-frontend/src/lib/modules/plan/components/plan-tree-editor/plan-tree-editor.component.ts

```diff
@@ -210,7 +210,9 @@ export class PlanTreeEditorComponent implements CustomComponent, PlanEditorStrat
   }
 
   handlePlanChange(): void {
-    this.planChange$.next(this.plan!);
+    this.planChange$.next({
+      ...this.plan!,
+    });
```

Comment:
well without changing the reference this _planEditService.plan$ would not emit, thus not propagating the changes back to the custom form and custom form input components
<img width="378" alt="image" src="https://user-images.githubusercontent.com/8954749/215772806-0708feeb-2887-454c-8b40-759470763a95.png">

Reply:
I've made some logging and see that value emits correctly, without this change.
For some reason `writeValue` not invoked in `step-custom-form-inputs`, but event the pipe is triggered extract the correct value.
I suggest to make pipe `stepCustomFormInputModel` impure (it will fix the issue), put a todo and investigate this case later

Reply:
@dvladir  I mean isnt it better the way I've already done it? I mean I did try to make the pipe impure and it does work, however I thought that a more optimal solution would be what I have done here, wouldnt it be bad for performance to just leave it impure (not that it would be felt but still worth considering)

Reply:
ok ultimately agreed for it to be impure for the time being